src/types/st-object.ts

```diff
@@ -1,11 +1,53 @@
 import { Field, FieldInstance } from "../enums";
 import { SerializedType, JsonObject } from "./serialized-type";
+import {
+  xAddressToClassicAddress,
+  isValidXAddress,
+} from "ripple-address-codec";
 import { BinaryParser } from "../serdes/binary-parser";
 import { BinarySerializer, BytesList } from "../serdes/binary-serializer";
 
-const OBJECT_END_MARKER = Buffer.from([0xe1]);
-const OBJECT_END_MARKER_NAME = "ObjectEndMarker";
-const OBJECT_FIELD_TYPE_NAME = "STObject";
+const OBJECT_END_MARKER_BYTES = Buffer.from([0xe1]);
+const OBJECT_END_MARKER = "ObjectEndMarker";
+const ST_OBJECT = "STObject";
+const DESTINATION = "Destination";
+const ACCOUNT = "Account";
+const SOURCE_TAG = "SourceTag";
+const DEST_TAG = "DestinationTag";
+
+/**
+ * Break down an X-Address into an account and a tag
+ *
+ * @param field Name of field
+ * @param xAddress X-Address corresponding to the field
+ */
+function handleXAddress(field: string, xAddress: string): JsonObject {
+  const decoded = xAddressToClassicAddress(xAddress);
+
+  let tagName;
+  if (field === DESTINATION) tagName = DEST_TAG;
+  else if (field === ACCOUNT) tagName = SOURCE_TAG;
+  else if (decoded.tag !== false)
+    throw new Error(`${field} cannot have an associated tag`);
+
+  return decoded.tag
```

Comment:
I think there's a rare edge case where a tag of `0` may fail to be used. While looking into this, I spotted a mistake in the ripple-address-codec README, and [opened a PR to fix it](https://github.com/ripple/ripple-address-codec/pull/114).

Checking that the tag is not `false` should fix the edge case:

```suggestion
  return decoded.tag !== false
```